spec/services/get_directions_service_spec.rb

```diff
@@ -49,20 +58,20 @@
   end
 
   describe '#call' do
-    context 'when validation passes' do
-      let(:store_coordinates) { [address.longitude, address.latitude] }
-
-      context 'when direction request succeeds' do
-        let(:request_success) { instance_double(Mapbox::GetDirectionsRequest, error?: false, response: result) }
-        let(:result) do
-          [{ 'duration' => 602.736,
-             'distance' => 3112.877,
-             'steps' =>
-               ['Drive southeast on Hồng Hà.',
-                'Turn right onto Đường Hoàng Minh Giám.'] }]
-        end
+    context 'when direction request succeeds' do
+      let(:request_success) { instance_double(Mapbox::GetDirectionsRequest, error?: false, response: result) }
+      let(:result) do
+        [{ 'duration' => 602.736,
+           'distance' => 3112.877,
+           'steps' =>
+             ['Drive southeast on Hồng Hà.',
+              'Turn right onto Đường Hoàng Minh Giám.'] }]
+      end
 
+      context 'when store coordinate exists in cache' do
         before do
+          allow(CacheManager).to receive(:fetch_value).and_return(store_coordinates.to_json)
```

Comment:
this line is duplicated a lot, why don't declare it in before in "call" context above, and in case the context need to returns value, just change it, instead of redeclare returns nil in each context